### PR TITLE
Enable TS strict mode and enforce it with GH actions

### LINKED ISSUE #89: Port to TypeScript
I'm uncomfortable writing increasingly large/complex software, which'll have multiple contributors, in a language without static types. But so far I've used TypeScript myself very little, and hit a number of snags with it, over in https://github.com/dabreegster/osm2svelte. Do we have any TypeScript experience on the incoming team? I'd like to have a discussion over how soon we should get used to using it, and what to watch out for.

### PR DESCRIPTION
**Pull request by dabreegster**
Closes #89.

In recent PRs, there have been some simple errors that TS would've caught, but we're not enforcing checks yet. My goal here is to catch new problems with as many compile-time warnings as possible. There are still some difficult cases for us to fix, but we can `@ts-ignore` them and treat them as TODOs.

### CODE REVIEW

File: src/lib/draw/Toolbox.svelte [hunk 1]
```diff
@@ -55,17 +55,19 @@
     "street-view": new EventHandler(),
   };
 
-  MapEvents.forEach((eventName) => {
+  for (let eventName of MapEvents) {
```
Comment (Pete-Y-CS) [inline src/lib/draw/Toolbox.svelte:58]: Out of interest why do we prefer `for` over `forEach`?

**Review by Pete-Y-CS — APPROVED** [review]

File: src/lib/sidebar/InterventionList.svelte [hunk 2]
```diff
@@ -17,8 +17,11 @@
       return feature.properties.planning?.name || "Untitled polygon";
     }
     if (schema == "v2") {
+      // TODO Revisit auto-generated enum types
```
Comment (dabreegster) [inline src/lib/sidebar/InterventionList.svelte:20]: Still needs more work to figure this out, but I  don't think it's worth blocking better checks

File: src/pages/App.svelte [hunk 3]
```diff
@@ -1,4 +1,5 @@
 <script lang="ts">
+  // @ts-ignore no declarations
```
Comment (dabreegster) [inline src/pages/App.svelte:2]: There's some way to define a .d.ts file to get TS declarations for external libraries, but haven't figured out exactly how, and it's not too important -- we're just calling this one init function

File: src/stores.ts [hunk 4]
```diff
@@ -11,7 +11,8 @@ import {
 import { type RouteInfo } from "./worker";
 
 // A global singleton, containing a loaded map
-// TODO | null. When we enable strictNullChecks, this'll become a problem
+// @ts-ignore TODO By construction, no components using the store should be
```
Comment (dabreegster) [inline src/stores.ts:14]: We could make all components using $map handle the possibility of not being loaded. https://github.com/dimfeld/svelte-maplibre/ does this, but it's complex. For now, I think carrying on with the assumption of it being loaded is OK

File: src/lib/draw/Toolbox.svelte (hunk 1 above)
Comment (dabreegster) [reply src/lib/draw/Toolbox.svelte:58]: Ah, will revert. I was trying a bunch of keyof/indexable type things that didn't work out. We use regular for loops in most places in the code, so it's default muscle memory for me

Comment (dabreegster) [reply src/lib/draw/Toolbox.svelte:58]: Discussed offline - in this case, not likely to share a separate function body anywhere. We'll prefer forEach / separate functions + functional methods especially in cases like #278 with the filter. Will go fix that up now

**Pull request merged**